editor/plugins/script_editor_plugin.h

```diff
@@ -308,6 +308,7 @@ class ScriptEditor : public PanelContainer {
 	int history_pos;
 
 	List<String> previous_scripts;
+	List<int> script_close_queue;
```

Comment:
Any reason why this can't be an argument to `_queue_close_tabs`?

Reply:
`_queue_close_tabs()` is interrupted by the dialog, which is asynchronous.

Reply:
Can we pass it through the connection arg list?

Reply:
Hmm, actually it should be possible.

Reply:
Ok so I tried the connection arg approach and it has a few problems. It can't take List, so I had to change the close queue to Vector, which is very wasteful for this case (especially the `pop_front()`). The worst part is that it doesn't actually work xd It compiles and closes the scripts, but in some cases some scripts are not being closed.

IMO having a helper variable in class is fine, if it simplifies the code and avoids problems. Changing it isn't really worth it.

Reply:
Ah that's shame, looks all good in that case.